pennylane/qnode.py

```diff
@@ -285,6 +285,7 @@ def construct(self, args, **kwargs):
             raise QuantumFunctionError('QNode._current_context must not be modified outside this method.')
         # generate the program queue by executing the quantum circuit function
         try:
+            Variable.kwarg_values = keyword_values
```

Comment:
Trying to understand, what is this assignment doing?

Reply:
When the `Operation` class queries for the value of a Variable, it is essentially replacing the variable identifier with the resulting value from the `Variable.kwarg_values` class attribute (shared by all Variable instances). Without this line here, the `Variable.kwarg_values` is never set, so on construction, the variable value cannot be returned (an exception is raised).